consensus/scp/src/slot.rs

```diff
@@ -494,10 +496,11 @@ impl<V: Value, ValidationError: Display> Slot<V, ValidationError> {
             // weight256 is the node's weight, scaled to 0..<max uint256>
             // (weight256 = <max uint256> * <num> / <denom>)
             let (num, denom) = self.weight(node_id);
-            let mut tmp = bigint::U512::from(bigint::U256::max_value());
-            tmp = tmp.saturating_mul(bigint::U512::from(num));
-            tmp = tmp.overflowing_div(bigint::U512::from(denom)).0;
-            let weight256 = bigint::U256::from(tmp);
+            let mut tmp = U512::from(U256::max_value());
+            tmp = tmp.saturating_mul(U512::from(num));
+            tmp /= U512::from(denom);
+            let weight256 = U256::try_from(tmp)
```

Comment:
How about we use `U256::full_mul`? It's description says `Multiplies two 256-bit integers to produce full 512-bit integer. Overflow is not possible.`

```suggestion
            let weight512: U512 = U256::max_value().full_mul(num.into()) / denom;
            let weight256 = U256::try_from(weight512)
```

Reply:
From my look at the underlying crate full mull is an overflowing multiplication instead of a saturating multiplication which is not what we want here.

Reply:
It's not overflowing because it puts the U256 operands in U512's before multiplying, and `2^256 * 2^256 == 2^512`

Reply:
I'm not sure how I feel about this change. I appreciate it's less code, but the code that's there now is resilient to, e.g. changes in `self.weight()` that would make `num` or `denom` bigger than `U256::max_value()`.

Reply:
I'm not sure what you mean. `weight()` returns `(u32, u32)`, which is a long way away from overflowing a `U256`. FWIW  `U256::full_mul()` only accepts `U256`, and `U256::from` is only implemented for narrower/smaller types.

With that said, I'm not attached to making this change.